core/src/components/datetime/datetime.tsx

```diff
@@ -1114,6 +1145,13 @@ export class Datetime implements ComponentInterface {
                 items={years}
                 value={workingParts.year}
                 onIonChange={(ev: CustomEvent) => {
+                  // Due to a Safari 14 issue we need to destroy
+                  // the intersection observer before we update state
+                  // and trigger a re-render.
+                  if (this.destroyCalendarIO) {
+                    this.destroyCalendarIO();
```

Comment:
What happens if we remove this? I tried testing in iOS 14 and I did not notice anything break.

Reply:
This specific code was to catch an issue when using the calendar month picker and changing months through that UI would re-render and cause the intersection observer to update the month back to the previous month. 

I noticed it prior by going from October -> September, then expanding the month picker and changing between months.

Was from this comment: https://github.com/ionic-team/ionic-framework/pull/24421#issuecomment-1023476866 (only applicable in Safari < 15).

Will prioritize time later tonight/early tomorrow to address the other quick scroll issue.

Reply:
We have identified this as a webkit bug, new issue being tracked here: #24683.

PR is ready to merge in the interim.